registry/component.go

```diff
@@ -448,3 +465,98 @@ func getSVGForComponent(model ModelCSV, component ComponentCSV) (colorSVG string
 	}
 	return
 }
+
+func getMinimalUICapabilitiesFromSchema() ([]capability.Capability, error) {
+    schema, err := schemas.Schemas.ReadFile("schemas/constructs/v1beta1/component/component.json")
+    if err != nil {
+        return nil, fmt.Errorf("failed to read component schema: %v", err)
+    }
+
+    allDefaults, err := extractCapabilitiesDefaultFromSchema(schema)
+    if err != nil {
+        return nil, fmt.Errorf("failed to extract capabilities from schema: %v", err)
+    }
+
+    if len(allDefaults) >= 3 {
+        lastThree := allDefaults[len(allDefaults)-3:]
+        return convertToCapabilityStructs(lastThree)
```

Comment:
you can use json.unmarshal to directly and safely convert the json to a proper struct . so you wont need to write this function